app/controllers/users_controller.rb

```diff
@@ -3,7 +3,20 @@ class UsersController < ApplicationController
   before_action :set_user, only: %i[show edit update]
 
   # GET /users/1 or /users/1.json
-  def show; end
+  def show
+    @my_datasets = []
+    @awaiting_datasets = []
+    @withdrawn_datasets = []
+    @datasets = []
+    @my_dashboard = current_user.id == @user.id
+    if @my_dashboard
+      @my_datasets = Dataset.my_datasets(current_user)
+      @awaiting_datasets = Dataset.admin_awaiting_datasets(current_user)
+      @withdrawn_datasets = Dataset.admin_withdrawn_datasets(current_user)
+    else
+      @datasets = Dataset.my_datasets(@user)
+    end
```

Comment:
Notice that the User show page serves two purposes now. 

If I am looking at my User show page then it "my dashboard" but if I am looking at another user's page then it just shows a list of datasets that user has created.